oauth2/src/main/java/com/linecorp/armeria/common/auth/oauth2/SerialFuture.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.auth.oauth2;
+
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.util.Exceptions;
+
+/**
+ * Executes submitted {@link Callable} actions asynchronously in sequence.
+ */
+public class SerialFuture {
```

Comment:
Could we remove `public` or move to `internal.common.auth.oauth2`?

Reply:
This is a generic utility class

Reply:
Yes, but I'd like to hide it from the public API and consider making it public when there are demands from users, because it's not an essential part of OAuth 2 support.

Reply:
moved to internal